frontend/src/utils/validation.ts

```diff
@@ -20,9 +20,16 @@ export const assertValidPublicKey = (pubKey: string, errorMsg?: string) => {
   }
 };
 
-export const assertValidEncryptionCount = (count: number, errorMsg?: string) => {
-  if (count < 0) {
-    throw new Error(errorMsg || 'Invalid count provided for encryption');
+export const assertValidEncryptionCount = (count: string, errorMsg?: string) => {
+  const error = new Error(errorMsg || 'Invalid count provided for encryption');
+  let bigNumberCount = BigNumber.from(-1);
+  try {
+    bigNumberCount = BigNumber.from(count);
+  } catch {
+    throw error;
+  }
+  if (bigNumberCount.lt(0)) {
+    throw error;
   }
```

Comment:
Can't we just do this?

```typescript
export const assertValidEncryptionCount = (count: BigNumberish, errorMsg?: string) => {
  if (BigNumber.from(count).lt(0)) {
    throw new Error(errorMsg || 'Invalid count provided for encryption');
  }
}
```

If the big number constructor fails I don't think we should hide that error and replace it with our own, cause that obscures what the actual issue was

Reply:
Good point

Reply:
https://github.com/ScopeLift/umbra-protocol/pull/494/commits/d7a4fc42cd05e201c10aa097798ab89aa6c7e858